diff --git a/chrome/browser/resources/settings/BUILD.gn b/chrome/browser/resources/settings/BUILD.gn
index 5e9e4797a9add24acad64387683ced62d130feaf..be074c0b9c18bab3298573318b33e95fa8f7bbe0 100644
--- a/chrome/browser/resources/settings/BUILD.gn
+++ b/chrome/browser/resources/settings/BUILD.gn
@@ -151,14 +151,6 @@ if (is_chromeos_ash) {
     extra_deps = [ ":build_ts" ]
   }
 
-  js_library("router") {
-    deps = [
-      "//ui/webui/resources/js:assert.m",
-      "//ui/webui/resources/js:load_time_data.m",
-    ]
-    externs_list = [ "$externs_path/metrics_private.js" ]
-  }
-
   js_library("setting_id_param_util") {
     sources = [ "$root_gen_dir/chrome/browser/resources/settings/tsc/setting_id_param_util.js" ]
     deps = [
@@ -169,6 +161,15 @@ if (is_chromeos_ash) {
   }
 }
 
+js_library("router") {
+  deps = [
+    "//ui/webui/resources/js:assert.m",
+    "//ui/webui/resources/js:load_time_data.m",
+  ]
+  externs_list = [ "$externs_path/metrics_private.js" ]
+}
+
+
 html_to_js("css_wrapper_files_old") {
   js_files = css_wrapper_files_old
 }
@@ -220,6 +221,8 @@ ts_library("build_ts") {
     ":preprocess",
     ":preprocess_generated",
   ]
+  in_files += brave_deps_chrome_browser_resources_settings_in_files extra_deps += brave_deps_chrome_browser_resources_settings_extra_deps
+
 }
 
 generate_grd("build_grd") {
@@ -308,4 +311,5 @@ generate_grd("build_grd") {
     manifest_files =
         filter_include(get_target_outputs(":build_ts"), [ "*.manifest" ])
   }
+  deps += [ "//brave/browser/resources/settings:resources" ]
 }
